gef.py

```diff
@@ -10521,30 +10525,34 @@ def parse_procfs_maps() -> Generator[Section, None, None]:
         return
 
     @staticmethod
-    def parse_gdb_info_sections() -> Generator[Section, None, None]:
+    def parse_gdb_info_proc_maps() -> Generator[Section, None, None]:
         """Get the memory mapping from GDB's command `maintenance info sections` (limited info)."""
-        stream = StringIO(gdb.execute("maintenance info sections", to_string=True))
 
-        for line in stream:
+        if GDB_VERSION < (11, 0):
+            raise AttributeError("Disregarding old format")
+
+        lines = (gdb.execute("info proc mappings", to_string=True) or "").splitlines()
+
+        # See expected format in tests/api/gef_memory.py:test_api_gef_memory_parse_info_proc_maps*
+        if len(lines) < 5:
+            raise AttributeError
+
+        # Format seems valid, iterate to generate sections
+        for line in lines[4:]:
             if not line:
                 break
 
-            try:
-                parts = [x for x in line.split()]
-                addr_start, addr_end = [int(x, 16) for x in parts[1].split("->")]
-                off = int(parts[3][:-1], 16)
-                path = parts[4]
-                perm = Permission.from_info_sections(parts[5:])
-                yield Section(page_start=addr_start,
-                              page_end=addr_end,
-                              offset=off,
-                              permission=perm,
-                              path=path)
-
-            except IndexError:
-                continue
-            except ValueError:
-                continue
+            parts = [x.strip() for x in line.split()]
+            addr_start, addr_end, offset, _ = list(map(lambda x: int(x, 16), parts[0:4]))
```

Comment:
Why not use consistent style?
```suggestion
            addr_start, addr_end, offset, _ = [int(x, 16) for x in parts[0:4]]
```
Also, if you aren't using the 4th part, why not do
```suggestion
            addr_start, addr_end, offset = [int(x, 16) for x in parts[0:3]]
```